pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go

```diff
@@ -195,13 +196,14 @@ func startProcessWithFile(t *testing.T, f *os.File) *exec.Cmd {
 
 // Check that we get (only) listening processes for all expected protocols.
 func TestBasic(t *testing.T) {
-	url := setupDiscoveryModule(t)
+	url, mockContainerProvider := setupDiscoveryModule(t)
+	mockContainerProvider.EXPECT().GetContainers(1*time.Minute, nil).AnyTimes()
 
 	var expectedPIDs []int
 	var unexpectedPIDs []int
 	expectedPorts := make(map[int]int)
 
-	var startTCP = func(proto string) {
+	startTCP := func(proto string) {
```

Comment:
Would be nice to avoid this and all the other code changes unrelated to the purpose of the PR.